src/lib/crypto/kimchi_backend/gadgets/ec_group.ml

```diff
@@ -216,7 +219,7 @@ let add (type f) (module Circuit : Snark_intf.Run with type field = f)
   let slope_squared_minus_x =
     Foreign_field.sub
       (module Circuit)
-      ~full:false slope_squared result_x curve.modulus
+      ~single:true slope_squared result_x curve.modulus
```

Comment:
Aha! The commented code did change! Don't do that.

Reply:
Oh but that's because of an old PR that was addressing some partial soundness fixes. It was accepted [here](https://github.com/MinaProtocol/mina/pull/13485) 

Perhaps should we merge that PR against develop directly, instead of adding on top of this one then?

Reply:
Seems like reverting is the best approach, and directly merge those to develop. As they are related, but not necessarily linked together.